README.md

```diff
@@ -1 +1,166 @@
-# Template
\ No newline at end of file
+# Ansible Playbook to deploy Starknet Node and Attestor Node
+
+This Ansible playbook automates the deployment and configuration of Starknet node and attestor node . It ensures that the necessary dependencies, configuration files, and services are properly set up and running.
+
+## Table of Contents
+
+- [Requirements](#requirements)
+- [Prerequisites](#prerequisites)
+- [Setup](#setup)
+- [Variables](#variables)
+- [Usage](#usage)
+
+## Requirements
+
+Before using this playbook, ensure the following requirements are met:
+
+1. **Ansible version**: Make sure you have Ansible 2.15+ installed.
+2. **SSH Access**: Passwordless SSH access to all target servers.
+3. **Python**: Python 3.x installed on the control node and all target hosts.
+4. **Privileges**: The user running the playbook must have sudo privileges on the target machines.
+
+## Prerequisites
+
+**Install HashiCorp Vault**
+
+This playbook relies on HashiCorp Vault to securely retrieve sensitive files, such as validator and node keys. Follow the [HashiCorp Vault Installation Guide](https://developer.hashicorp.com/vault/tutorials/getting-started/getting-started-install) to set up Vault on your infrastructure.
+
+**Note on Secrets Management**
+
+The playbook dynamically retrieves secret environment file from HashiCorp Vault. The keys are expected to follow a structured path format:
+`<environment>/<project>/<organization>/<type>/<file_name>`
+For example:
+`mainnet/starknet/encapsulate/attestor/starknet-attestor.secrets.env`
```

Comment:
Lets directly add a link to that file here for the file structure, also include a dummy value in the file